Handle `state clean config -n` correctly and report abortion of clean process.

mitchell-as opened the pull request:
<!--do not remove this marker, its needed to replace info when ticket title is updated -->
<!--jira-description-action-hidden-marker-start-->

<table>
<td>
  <a href="https://activestatef.atlassian.net/browse/DX-1239" title="DX-1239" target="_blank"><img alt="Bug" src="https://activestatef.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/10303?size=medium" />DX-1239</a>  `state clean config -n` does nothing
  </td></table>
  <br />
 

<!--jira-description-action-hidden-marker-end-->

Naatan commented on internal/runners/clean/config.go:
```diff
@@ -53,12 +54,13 @@ func (c *Config) Run(params *ConfigParams) error {
 
 	if !params.Force {
-		ok, err := c.confirm.Confirm(locale.T("confirm"), locale.T("clean_config_confirm"), new(bool))
+		defaultChoice := params.NonInteractive
+		ok, err := c.confirm.Confirm(locale.T("confirm"), locale.T("clean_config_confirm"), &defaultChoice)
```
This is already handled: https://github.com/ActiveState/cli/blob/52511befe457f17b63048fdc07b78be9d50c861f/internal/prompt/prompt.go#L149

You can just pass `p.BoolP(true)`. The default is used when ran under non-interactive mode.

In fact I think you can reduce this PR to pretty much this one change. You shouldn't need to pass NonInteractive to the runner.

mitchell-as replied:
But we want the default to be `false` if running interactively. If I pass `p.BoolP(true)`, then running `state clean config` prompts me with `Y/n` (Y) as the default. What am I missing?

Naatan replied:
> But we want the default to be false if running interactively

I think you're asserting that based on what the code currently does? I'm not seeing anything on the Jira ticket about this.

Imo the default should just be true, regardless of what mode you're running under. This is how we've handled all non-interactive prompts so far. The reason why there are some prompts with the wrong default is because the prompts preceded the non-interactive mechanic, so not all of them are well curated towards the non-interactive mechanic.

Naatan reviewed (changes requested):
https://github.com/ActiveState/cli/pull/2106#discussion_r970034241

mitchell-as commented:
Okay, I changed the default to be Yes. The other changes are still valid. One locale id was incorrect, updated another to be more descriptive, for consistency we should return input error when confirmation is not given, and updated an affected unit test.
